source/slang-rt/slang-rt.h

```diff
@@ -0,0 +1,22 @@
+// slang-rt.h
+#ifndef SLANG_RT_H
+#define SLANG_RT_H
+
+#include "../core/slang-string.h"
+#include "../core/slang-smart-pointer.h"
+
+#ifdef SLANG_RT_DYNAMIC_EXPORT
+#    define SLANG_RT_API SLANG_DLL_EXPORT
+#else
+#    define SLANG_RT_API
+#endif
+
+extern "C"
+{
+    SLANG_RT_API void SLANG_MCALL _slang_rt_abort(Slang::String errorMessage);
+    SLANG_RT_API void* SLANG_MCALL _slang_rt_load_dll(Slang::String modulePath);
+    SLANG_RT_API void* SLANG_MCALL
+        _slang_rt_load_dll_func(void* moduleHandle, Slang::String modulePath);
+}
```

Comment:
This file probably merits a comment of some kind to clarify what exactly `slang-rt` is meant to encapsulate in scope.

Also: looking at these operations, I assume we probably want the `String`s to be passed by-reference (or maybe even by-pointer, to be even more explicit about ABI considerations).

I expect that in the long run we will need to think carefully about how we carve up functionality that generated programs need across:

* The "standard library" code, which is currently written in Slang, compiled to our IR, and subsequently gets emitted as part of the generated code of programs using those routines (effectively getting statically linked into output files via duplication of the code).

* The "prelude" code, which is written in the downstream non-Slang language, and gets prepended to the generated code of programs (such that it is *also* statically linked by duplication)

* The "runtime" code, which differs in practice by not being included directly into the code we generate, and is instead linked to that code further downstream (in this case by the runtime library loader of the host platform).

We should probably be looking to unify the handling of these different cases down the line, so that we have a clear mental model like:

* The standard library declares all the operations that are visible to client code (including runtime operations that we might not always want them to access directly), as well as any lower-level operations that are used to implement the standard library itself. Some operations in the stdlib will be declared but not defined, and those operations are likely to have their definitions provided by the prelude and/or runtime for a target. While the stdlib currently gets compiled into generated output code from the Slang compiler, it should eventually be possible for parts of the stdlib to be precompiled into binaries for appropriate targets (binaries such as the runtime library).

* The prelude is simply used as a utility to generated code; it provides declarations and (inline) definitions for those operations that are needed by generated code, and which are best implemented by inlining into the output for a given target. In some cases the prelude may simply declare operations that are actually provided by the runtime, or may provide inline implementations that forward to the runtime.

* The runtime library provides the target-specific (and platform-specific) implementations of those services that are needed by generated code and that can be best provided by a pre-built binary that is deployed with the compiler.

I fully understand that for now we are unlikely to have any "runtime library" for most of our GPU targets, but we should start thinking about how to architect things for a world where these different pieces are designed to clearly work together (rather than the current situation where they sometimes come across as competing solutions to the same basic problems).